### PR TITLE
New Dialog UI for webapp creation

### PR DESCRIPTION
**Pull request by Flanker32**
New Dialog UI for webapp creation
* Create webapp with existing resource group and app service plan
image

* Create webapp with new resource group and app service plan
image

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/slimui/creation/WebAppCreationDialog.java [hunk 1]
```diff
@@ -0,0 +1,483 @@
...
+                if (Comparing.equal(location.name(), DEFAULT_REGION.name())) {
+                    cbLocation.setSelectedItem(location);
+                }
+            });
+        pack();
+    }
+
+    @Override
+    public void fillPricingTier(@NotNull List<PricingTier> prices) {
+        cbPricing.removeAllItems();
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/slimui/creation/WebAppCreationDialog.java]: duplicode as fillWebContainer...

**Comment by Eskibear** [comment]
Is it on purpose to highlight "Runtime" by a different UI ?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/slimui/creation/WebAppCreationDialog.java [hunk 2]
```diff
@@ -0,0 +1,483 @@
...
+            webAppConfiguration.setCreatingAppServicePlan(false);
+            AppServicePlan appServicePlan = (AppServicePlan) cbExistAppServicePlan.getSelectedItem();
+            if (appServicePlan != null) {
+                webAppConfiguration.setAppServicePlanId(appServicePlan.id());
+            }
+        }
+        // runtime
+        if (rdoLinuxOS.isSelected()) {
+            webAppConfiguration.setOS(OperatingSystem.LINUX);
+            RuntimeStack linuxRuntime = (RuntimeStack) cbRuntimeStack.getSelectedItem();
```
Comment (zmssp) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/slimui/creation/WebAppCreationDialog.java]: If linuxRuntime will be null, here cannot add cast.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/slimui/creation/WebAppCreationDialog.java [hunk 3]
```diff
@@ -0,0 +1,483 @@
...
+            if (linuxRuntime != null) {
+                webAppConfiguration.setStack(linuxRuntime.stack());
+                webAppConfiguration.setVersion(linuxRuntime.version());
+            }
+        }
+        if (rdoWindowsOS.isSelected()) {
+            webAppConfiguration.setOS(OperatingSystem.WINDOWS);
+            JdkModel jdkModel = (JdkModel) cbJdkVersion.getSelectedItem();
+            if (jdkModel != null) {
+                webAppConfiguration.setJdkVersion(jdkModel.getJavaVersion());
```
Comment (zmssp) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/slimui/creation/WebAppCreationDialog.java]: If jdkModel may be null, here cannot cast directly

**Comment by Flanker32** [comment]
> Is it on purpose to highlight "Runtime" by a different UI ?

Yes, the origin design is that user could only care about runtime configurations and appservice plan/resource group are default to fold. But after discussion, the app service plan / resoure group should also be printed as user should know whether they are creating new resources, so should we remove the decorator of runtime?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/slimui/creation/WebAppCreationDialog.java (hunk 1 above)
Comment (Flanker32) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/slimui/creation/WebAppCreationDialog.java]: yes, use new function `fillCombobox` to do fill operation

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/slimui/creation/WebAppCreationDialog.java (hunk 3 above)
Comment (Flanker32) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/slimui/creation/WebAppCreationDialog.java]: added null check, thanks~

**Comment by Flanker32** [comment]
Update the UI design, and here is the new screenshot
* Create webapp with existing resource group and app service plan
image

* Create webapp with new resource group and app service plan
image

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/slimui/creation/WebAppCreationDialog.java [hunk 4]
```diff
@@ -0,0 +1,434 @@
...
+    public void fillJdkVersion(@NotNull List<JdkModel> jdks) {
+        fillCombobox(this, cbJdkVersion, jdks, DEFAULT_WINDOWS_JAVAVERSION);
+    }
+
+    @Override
+    public void fillLinuxRuntime(@NotNull List<RuntimeStack> linuxRuntimes) {
+        fillCombobox(this, cbRuntimeStack, linuxRuntimes, DEFAULT_LINUX_RUNTIME);
+    }
+
+    public static <T> void fillCombobox(Window window, JComboBox<T> comboBox, List<T> values, T defaultValue) {
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/slimui/creation/WebAppCreationDialog.java]: why is it public for external package?

Comment (Flanker32) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/slimui/creation/WebAppCreationDialog.java]: Plan to make it as an common method before, will fix it as other fill operation are more complex.

**Pull request merged**